plugins/out_s3/s3.c

```diff
@@ -1239,6 +1419,19 @@ static struct multipart_upload *create_upload(struct flb_s3 *ctx,
     m_upload->init_time = time(NULL);
     mk_list_add(&m_upload->_head, &ctx->uploads);
 
+    /* Update file and increment index value right before request */
+    if (ctx->key_fmt_has_seq_index) {
+        ctx->seq_index++;
```

Comment:
If the create multipart upload request fails- will seq_index be decremented? I see the decrement code for put_object, I can't find it for multipart.

Reply:
The `seq_index` decrement is in the multipart section of `upload_data` after `create_multipart_upload` is called. I had to do it this way because error handling for multipart upload requests are handled outside of `create_upload`.